### PR TITLE
Introduce format_plural

### PR DESCRIPTION
**Pull request by tsusanka**
I'm filing a PR to find out what you think before I reformat the rest.

- I'm not sure about the named parameters, should we use unnamed instead?
- Do we want to support irregularities? Or let's do it later when needed? @prusnak gave some hints from [django](https://github.com/django/django/blob/8f6860863e34cb1cbe24161f1c4e7c79007e93dc/django/template/defaultfilters.py#L851-L890).
- Fix support for 0 (it should use plural).

### CODE REVIEW

**Comment by tsusanka** [comment]
This was supposed to be a Draft PR, but I've forgot and it [can't](https://stackoverflow.com/questions/55070723/how-do-i-change-from-a-pr-to-a-draft-pr-at-github) be changed :(.

**Comment by prusnak** [comment]
I like the change!

I suggest to change the main inflexing code to this

``` python
if count == 0 or count > 1:
    if plural[-1] == "y":
        plural = plural[:-1] + "ies"
    elif plural[-1] in "hsxz":
        plural = plural + "es"
    else:
        plural = plural + "s"
```

File: core/src/trezor/strings.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    Example:
+    >>> format_plural("We need %(count)s more %(plural)s", 3, "share")
+    >>> 'We need 3 more shares'
+    >>> format_plural("We need %(count)s more %(plural)s", 1, "share")
+    >>> 'We need 1 more share'
+    """
+    if not all(s in string for s in ("%(count)", "%(plural)")):
+        # string needs to have %(count) and %(plural) inside
+        raise ValueError()
```
Comment (matejcik) [inline core/src/trezor/strings.py]: if there are no arguments, it should just be `raise ValueError` without parens

**Comment by matejcik** [comment]
I think named parameters are better than unnamed for this usecase -- consider: `"number of %(plural)s: %(count)s`"

but perhaps with new-style formatting?
`"We need {count} more {plural}"` is nicer than `%(plural)s` IMHO.
(and then `s.format(count=count, plural=plural)`)

This is of course somewhat of a bikeshed, but IIRC we were supposed to move to new-style formatting everywhere, @jpochyla?

Otherwise looks nice, also agreed with @prusnak's impl

File: core/src/trezor/strings.py (hunk 1 above)
Comment (tsusanka) [reply core/src/trezor/strings.py]: I always forget that. Is the reason the without () I'm not creating the object and therefore saving an allocation?

Comment (matejcik) [reply core/src/trezor/strings.py]: the reason is Python style guide :) but yes, what you said is also true

File: core/src/trezor/strings.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    if count == 0 or count > 1:
+        if plural[-1] == "y":
+            plural = plural[:-1] + "ies"
+        elif plural[-1] in "hsxz":
+            plural = plural + "es"
+        else:
+            plural = plural + "s"
+
+    return string.format(count=str(count), plural=plural)
```
Comment (matejcik) [inline core/src/trezor/strings.py]: the `str(count)` is superfluous

File: core/src/trezor/strings.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+
+def format_plural(string: str, count: int, plural: str):
+    """
+    Adds plural form to a string based on `count`.
+    !! Does not work with irregular words !!
+
+    Example:
+    >>> format_plural("We need {count} more {plural}", 3, "share")
+    >>> 'We need 3 more shares'
```
Comment (jpochyla) [inline core/src/trezor/strings.py]: Little nit: output (result value) should be without `>>>`.

File: core/src/trezor/strings.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    >>> format_plural("We need {count} more {plural}", 1, "share")
+    >>> 'We need 1 more share'
+    >>> format_plural("{count} {plural}", 4, "candy")
+    >>> '4 candies'
+    """
+    if not all(s in string for s in ("{count}", "{plural}")):
+        # string needs to have {count} and {plural} inside
+        raise ValueError
```
Comment (jpochyla) [inline core/src/trezor/strings.py]: I'm not sure this is really needed.

Comment (tsusanka) [reply core/src/trezor/strings.py]: It helps to use the function correctly, doesn't it? But maybe it's not worth it

**Comment by tsusanka** [comment]
I've fixed the code based on the comments, but we might want to merge this after #431 because it's a great use case for that test.

**Comment by matejcik** [comment]
fished out and rebased this one. UI tests seem to think that the change is correct

File: core/tests/test_trezor.strings.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            (-123456, 3, "-123.456"),
+            (-4242, 7, "-0.0004242"),
+        ]
+        for v in VECTORS:
+            self.assertEqual(strings.format_amount(v[0], v[1]), v[2])
+
+
+    def test_format_plural(self):
+        VECTORS = [
+            ("We need {count} more {plural}", 3, "share", "We need 3 more shares"),
```
Comment (prusnak) [inline core/tests/test_trezor.strings.py]: Please add testcase for words ending with `hsxz`, such as "hash", etc.

**Pull request merged**